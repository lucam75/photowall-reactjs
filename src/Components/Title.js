import React, {Component} from 'react'
import { Link } from 'react-router-dom';

class Title extends Component{

    render(){
        
    return <div className="row">
                <div className="col-12">
                    <nav className="navbar navbar-light bg-light">
                        <h1><Link to="/">{this.props.title}</Link></h1>
                        {
                            !this.props.user.email ?
                                <Link className="add-photo btn btn-primary" to="/AddPhoto">Add Photo</Link> : ''
                            
                        }
                        {
                        !this.props.user.email? <Link className="" to="/Login"><span className="logInIcon"></span></Link> : <span>{this.props}</span>
                        }
                    </nav>
                </div>
            </div>
    }
}

export default Title;