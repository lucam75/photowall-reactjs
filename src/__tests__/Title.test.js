import React, {Component} from 'react'
import { create } from "react-test-renderer";
import {BrowserRouter} from 'react-router-dom'
import Title from '../Components/Title'

describe("Title component", () => {
    test('Title matches the snapshot', () => {
        const component = create(<BrowserRouter><Title /></BrowserRouter>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    test("it shows the expected text when is rendered", () => {
      /*const component = create(<BrowserRouter><Title title="Photowall" /></BrowserRouter>);
      const instance = component.getInstance();
      expect(instance.props('title')).toBe("Photowall");*/
    });
  });