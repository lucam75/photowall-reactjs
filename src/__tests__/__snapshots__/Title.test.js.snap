// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component Title matches the snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <nav
      className="navbar navbar-light bg-light"
    >
      <h1>
        <a
          href="/"
          onClick={[Function]}
        />
      </h1>
      
      <a
        className=""
        href="/Login"
        onClick={[Function]}
      >
        <span
          className="logInIcon"
        />
      </a>
    </nav>
  </div>
</div>
`;

exports[`Title is rendered 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <nav
      className="navbar navbar-light bg-light"
    >
      <h1>
        <a
          href="/"
          onClick={[Function]}
        />
      </h1>
      
      <a
        className=""
        href="/Login"
        onClick={[Function]}
      >
        <span
          className="logInIcon"
        />
      </a>
    </nav>
  </div>
</div>
`;

exports[`Title it matches the snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <nav
      className="navbar navbar-light bg-light"
    >
      <h1>
        <a
          href="/"
          onClick={[Function]}
        />
      </h1>
      
      <a
        className=""
        href="/Login"
        onClick={[Function]}
      >
        <span
          className="logInIcon"
        />
      </a>
    </nav>
  </div>
</div>
`;

exports[`Title matches the snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <nav
      className="navbar navbar-light bg-light"
    >
      <h1>
        <a
          href="/"
          onClick={[Function]}
        />
      </h1>
      
      <a
        className=""
        href="/Login"
        onClick={[Function]}
      >
        <span
          className="logInIcon"
        />
      </a>
    </nav>
  </div>
</div>
`;
